src/language/source.js

```diff
@@ -8,18 +8,26 @@
  *  of patent rights can be found in the PATENTS file in the same directory.
  */
 
+type Location = {
+  line: number,
+  column: number,
+};
+
 /**
- * A representation of source input to GraphQL. The name is optional,
- * but is mostly useful for clients who store GraphQL documents in
- * source files; for example, if the GraphQL input is in a file Foo.graphql,
- * it might be useful for name to be "Foo.graphql".
+ * A representation of source input to GraphQL.
+ * `name` and `location` are optional. They are useful for clients who store
+ * GraphQL documents in source files; for example, if the GraphQL input starts
+ * at line 40 in a file named Foo.graphql, it might be useful for name to be
+ * "Foo.graphql" and location to be `{ line: 40, column: 0 }`.
  */
 export class Source {
   body: string;
   name: string;
+  location: Location;
 
-  constructor(body: string, name?: string): void {
+  constructor(body: string, name?: string, location?: Location): void {
     this.body = body;
     this.name = name || 'GraphQL request';
+    this.location = location || { line: 1, column: 0 };
```

Comment:
For some reason lines are 1-indexed and columns are 0-indexed. I just looked at what Emacs prints in the modeline when I wrote this, but just confirmed with the Babylon docs: https://github.com/babel/babylon/blob/master/ast/spec.md#node-objects